CRM/Core/BAO/OptionValue.php

```diff
@@ -206,33 +217,30 @@ public static function add(&$params, $ids = array()) {
       CRM_Core_DAO::executeQuery($query, $p);
     }
 
-    // CRM-13814 : evalute option group id
-    if (!array_key_exists('option_group_id', $params) && !empty($ids['optionValue'])) {
-      $groupId = CRM_Core_DAO::getFieldValue('CRM_Core_DAO_OptionValue',
-        $ids['optionValue'], 'option_group_id', 'id'
-      );
-    }
-    else {
-      $groupId = $params['option_group_id'];
+    if (empty($params['domain_id']) && in_array($groupName, CRM_Core_OptionGroup::$_domainIDGroups)) {
+      $optionValue->domain_id = CRM_Core_Config::domainID();
     }
 
-    $groupName = CRM_Core_DAO::getFieldValue('CRM_Core_DAO_OptionGroup',
-      $groupId, 'name', 'id'
-    );
-    if (empty($ids['optionValue']) && empty($params['id']) && !empty($params['value'])) {
-      $domainSpecifc = in_array($groupName, CRM_Core_OptionGroup::$_domainIDGroups) ? TRUE : FALSE;
+    $groupsSupportingDuplicateValues = ['languages'];
+    if (!$id && !empty($params['value'])) {
       $dao = new CRM_Core_DAO_OptionValue();
-      $dao->value = $params['value'];
-      $dao->option_group_id = $groupId;
+      if (!in_array($groupName, $groupsSupportingDuplicateValues)) {
```

Comment:
I think there is no need to have separate variable ```$groupsSupportingDuplicateValues``` as it only used in this condition. So we simply change it to 
```patch
- $groupsSupportingDuplicateValues = ['languages'];
..
- if (!in_array($groupName, $groupsSupportingDuplicateValues)) {
+ if (!in_array($groupName, array('languages)) {
```

Reply:
The reason I made it an array is I suspect there will be more....  - also the var name provides an explanation that hard-coding doesn't

Reply:
oh ok, well then I am merging it then